packages/hierarchy-builder/src/test/internal/operators/Grouping.test.ts

```diff
@@ -0,0 +1,207 @@
+              byBaseClasses: {
+                fullClassNames: ["TestSchema.TestParentClassA", "TestSchema.TestParentClassAA"],
+                hideIfNoSiblings: true,
+              },
+              byLabel: true,
+              byClass: true,
+            },
+          },
+        }),
+      ];
+
+      const result = await getObservableResult(
+        from(nodes).pipe(
+          createGroupingOperator(metadata, [
+            async (allNodes) => {
+              return { grouped: [], ungrouped: allNodes, groupingType: "label" } as GroupingHandlerResult;
+            },
+            async (allNodes) => {
+              return { grouped: [], ungrouped: allNodes, groupingType: "class" } as GroupingHandlerResult;
+            },
+          ]),
+        ),
+      );
+      expect(applyGroupingHidingParamsStub.callCount).to.eq(2);
+      expect(result).to.deep.eq(nodes);
+    });
+
+    it("runs grouping handlers in provided order", async () => {
+      const nodes = [
+        createTestNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.A", id: "0x1" }] },
+          label: "1",
+          params: {
+            grouping: {
+              byClass: true,
+              byLabel: true,
+            },
+          },
+        }),
+        createTestNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.B", id: "0x2" }] },
+          label: "1",
+          params: {
+            grouping: {
+              byLabel: true,
+            },
+          },
+        }),
+      ];
+
+      const result = await getObservableResult(
+        from(nodes).pipe(
+          createGroupingOperator(metadata, [
+            async (allNodes) => {
+              return {
+                grouped: [
+                  {
+                    label: "TestSchema A",
+                    key: {
+                      type: "class-grouping",
+                      class: {
+                        name: "TestSchema A",
+                      },
+                    },
+                    children: [allNodes[0]],
+                  },
+                ],
+                ungrouped: [allNodes[1]],
+                groupingType: "class",
+              } as GroupingHandlerResult;
+            },
+            async (allNodes) => {
+              return {
+                grouped: [
+                  {
+                    label: "1",
+                    key: {
+                      type: "label-grouping",
+                      label: "1",
+                    },
+                    children: allNodes.filter((node) => !HierarchyNode.isClassGroupingNode(node)),
+                  },
+                ],
+                ungrouped: allNodes.filter((node) => HierarchyNode.isClassGroupingNode(node)),
+                groupingType: "label",
+              } as GroupingHandlerResult;
+            },
+          ]),
+        ),
+      );
+      expect(applyGroupingHidingParamsStub.callCount).to.eq(3);
```

Comment:
should we also check what arguments were passed to each call of this stub and also the grouping handlers?

Reply:
Added checks.